### PR TITLE
Feature/removing osf dependencies

### PR DESCRIPTION
**Pull request by cwisecarver**
@sloria I built this to make the example app work. It makes the user model compliant with django along with moving some dependencies that exist in website into this app. I'm going to merge it locally so I can start some work in moving Node and it's brethren to being based off of the TypedModel class. If you want to give it a once over tomorrow and let me know if it's disastrous based on what changes you've made I'd appreciate it. TIA.

### CODE REVIEW

File: osf_models/initial_data.json [hunk 1]
```diff
@@ -0,0 +1,819 @@
+[
```
Comment (sloria) [inline osf_models/initial_data.json]: Is this file necessary?

**Comment by sloria** [comment]
@cwisecarver Is there an companion OSF branch that is compatible with these changes, e.g. with the exception imports changed?

File: osf_models/utils/tokens.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import httplib as http
+import functools
+import jwt
+from flask import request
+
+from framework.exceptions import HTTPError
+
+from website import settings
+from website.tokens import handlers
+from website.tokens.exceptions import TokenHandlerNotFound
```
Comment (sloria) [inline osf_models/utils/tokens.py]: Won't these dependencies need to be moved over as well?

**Comment by sloria** [comment]
Looking good on first glance. 

We discussed this briefly on FD, but we might consider an alternative approach: instead of moving all the dependencies and all the business logic to this repo, I wonder if it would be (1) less work and (2) easier to integrate into the OSF if we kept that code in the OSF and added a "services" layer that encapsulated behavior. The models in this repo would have almost no behavior; they would just define the schemas. Thoughts?

File: osf_models/initial_data.json (hunk 1 above)
Comment (cwisecarver) [inline osf_models/initial_data.json]: It was mostly for the example django project I made for teams to coordinate their django apps with ours but I think Preprints is going in another direction. I'll cut this for now. It's easily recreated if we need it. Just a default admin user and a bunch of django contenttypes.

File: osf_models/utils/tokens.py (hunk 2 above)
Comment (cwisecarver) [inline osf_models/utils/tokens.py]: Yes, I think by commenting out all the sanctions stuff this was disused so it never threw an error so I didn't catch it.

**Comment by cwisecarver** [comment]
I think integrating it into the OSF might make it easier. I'm not 100% sold on the services layer. Most of the service layers I've ever seen implemented ended up being a super thin layer of classes that just ended up making more code without any benefits. Maybe we could chat on flowdock more about how you see that working. I have less concern if it's something temporary to patch the django models into the flask app and the current implementation of v2. But if it's something that would permanently sit between the v2 API and the django models I think we'd be stepping away from the defined django patterns and possibly down a rabbit hole that might burn us later.

I think most of the value in this branch at this point (now that the example app intended for concurrent development is probably not going to be used right away, if ever) is just making the OSFUser model match the interface required for user models in django.

**Comment by sloria** [comment]
@cwisecarver Should this be closed?

**Comment by cwisecarver** [comment]
Yup. Closing.
